pilot/platform/kube/inject/inject.go

```diff
@@ -505,16 +505,29 @@ func intoObject(c *Config, in interface{}) (interface{}, error) {
 	// `in` is a pointer to an Object. Dereference it.
 	outValue := reflect.ValueOf(out).Elem()
 
-	templateValue := outValue.FieldByName("Spec").FieldByName("Template")
-	// `Template` is defined as a pointer in some older API
-	// definitions, e.g. ReplicationController
-	if templateValue.Kind() == reflect.Ptr {
-		templateValue = templateValue.Elem()
+	var objectMeta *metav1.ObjectMeta;
+	var templateObjectMeta *metav1.ObjectMeta;
+	var templatePodSpec *v1.PodSpec;
+	// CronJobs have JobTemplates in them, instead of Templates, so we
+	// special case them.
+	if outValue.FieldByName("TypeMeta").FieldByName("Kind").String() != "CronJob" {
+		templateValue := outValue.FieldByName("Spec").FieldByName("Template")
+		// `Template` is defined as a pointer in some older API
+		// definitions, e.g. ReplicationController
+		if templateValue.Kind() == reflect.Ptr {
+			templateValue = templateValue.Elem()
+		}
+		objectMeta = outValue.FieldByName("ObjectMeta").Addr().Interface().(*metav1.ObjectMeta)
+		templateObjectMeta = templateValue.FieldByName("ObjectMeta").Addr().Interface().(*metav1.ObjectMeta)
+		templatePodSpec = templateValue.FieldByName("Spec").Addr().Interface().(*v1.PodSpec)
+	} else {
```

Comment:
No need to use reflection for a specific kind. You can conditional type-assert to CronJob and pick-off the required fields directly, e.g.

```golang
    if job, ok := out.(*v1alpha1.CronJob); ok {
        objectMeta = &job.ObjectMeta
        templateObjectMeta = &job.Spec.JobTemplate.ObjectMeta
        templatePodSpec = &job.Spec.JobTemplate.Spec.Template
    } else {
        // use reflection                                                                                                                                                           
    }
```

Reply:
Thanks! Not used to Go -- that does sound cleaner. Done.